### PR TITLE
CORE-1601: made the username suffix configurable

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

**Review by johnworth — APPROVED** [review]
I didn't see anything obviously wrong with this. :shipit:

**Comment by slr71** [comment]
Thanks for the review! I'll wait until I can merge the `apps` service changes before merging this one.

**Review by sri2k1us — CHANGES_REQUESTED** [review]
So far so good but I think this needs some more changes. First, for the config to be available on the client side, the config must be read in using [next.config.js](https://github.com/cyverse-de/sonora/blob/master/next.config.js) and set up in [pages/_apps.js](https://github.com/cyverse-de/sonora/blob/625be0bae467f622efaa079efca6f69de210058a/src/pages/_app.js#L206). 
Also I found a typo that breaks the server start-up.

File: config/default.yaml
Comment (slr71) [reply config/default.yaml:90]: D'oh! Thanks for catching that.

**Comment by slr71** [comment]
I'm a little confused about the requested change for reading the configuration. It was working when I tested it yesterday using `npm run dev` is there another test that I should be doing to make sure that the configuration will always be available? Or were you just saying that there should be a constant available for this setting rather than accessing the configuration object?

**Comment by sri2k1us** [comment]
> I'm a little confused about the requested change for reading the configuration. It was working when I tested it yesterday using `npm run dev` is there another test that I should be doing to make sure that the configuration will always be available? Or were you just saying that there should be a constant available for this setting rather than accessing the configuration object?

I used `nmp run dev` too but I set up the config for suffix to be something like `foo.org` and it didn't show up in the `getUserNameSuffix` function. I can show you what I did in zoom call.

**Comment by slr71** [comment]
> > I'm a little confused about the requested change for reading the configuration. It was working when I tested it yesterday using `npm run dev` is there another test that I should be doing to make sure that the configuration will always be available? Or were you just saying that there should be a constant available for this setting rather than accessing the configuration object?
> 
> I used `nmp run dev` too but I set up the config for suffix to be something like `foo.org` and it didn't show up in the `getUserNameSuffix` function. I can show you what I did in zoom call.

The Zoom call helped. Thanks!

The problem was that I didn't realize that the JSON object for the configuration is being reconstructed after being read in from the configuration file. The reason it was working was that I had the code in place for making the configuration setting optional, and that was in place. I'm a little embarrassed for not thinking about that, but that was what happened.

**Comment by sri2k1us** [comment]
> > > I'm a little confused about the requested change for reading the configuration. It was working when I tested it yesterday using `npm run dev` is there another test that I should be doing to make sure that the configuration will always be available? Or were you just saying that there should be a constant available for this setting rather than accessing the configuration object?
> > 
> > 
> > I used `nmp run dev` too but I set up the config for suffix to be something like `foo.org` and it didn't show up in the `getUserNameSuffix` function. I can show you what I did in zoom call.
> 
> The Zoom call helped. Thanks!
> 
> The problem was that I didn't realize that the JSON object for the configuration is being reconstructed after being read in from the configuration file. The reason it was working was that I had the code in place for making the configuration setting optional, and that was in place. I'm a little embarrassed for not thinking about that, but that was what happened.

No need to be embarrassed. As I said in the call, When we set up nextjs, we had a lot of difficulty figuring this out. Any way, constructing the config object `_app.js`  is one aspect but the main thing is reading the config file in `next.config.js` is step we all forget.  Here is the link to nextjs documentation - https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration

**Review by sri2k1us — APPROVED** [review]
LGTM :shipit:

File: src/components/analyses/listing/TableView.js
Comment (psarando) [inline src/components/analyses/listing/TableView.js:187]: Looks like debug logging that can be removed now.

File: src/components/analyses/toolbar/AnalysesDotMenu.js
Comment (psarando) [inline src/components/analyses/toolbar/AnalysesDotMenu.js:301]: Looks like debug logging that can be removed now.

File: src/components/utils/getUserName.js
Comment (psarando) [inline src/components/utils/getUserName.js:29]: Since we're naming this as a `suffix` explicitly now, I wonder if we should use `endsWith` instead of `includes`?
```suggestion
    return userNameWithDomain && userNameWithDomain.endsWith(suffix)
```

**Review by psarando — APPROVED** [review]
LGTM 👍

**Comment by slr71** [comment]
Thanks for the reviews!

**Pull request merged**
